app/code/Magento/ConfigurableProduct/Model/Product/VariationHandler.php

```diff
@@ -47,49 +54,59 @@ class VariationHandler
      */
     private $attributes;
 
+    /**
+     * @var \Magento\ConfigurableProduct\Model\Product\ChildrenWeightResolver
+     */
+    private $childrenWeightResolver;
+
     /**
      * @var \Magento\CatalogInventory\Api\StockConfigurationInterface
      * @deprecated 100.1.0
      */
     protected $stockConfiguration;
 
     /**
-     * @param Type\Configurable $configurableProduct
-     * @param \Magento\Eav\Model\Entity\Attribute\SetFactory $attributeSetFactory
-     * @param \Magento\Eav\Model\EntityFactory $entityFactory
-     * @param \Magento\Catalog\Model\ProductFactory $productFactory
-     * @param \Magento\CatalogInventory\Api\StockConfigurationInterface $stockConfiguration
-     * @param \Magento\Catalog\Model\Product\Gallery\Processor $mediaGalleryProcessor
+     * @param Type\Configurable           $configurableProduct
+     * @param SetFactory                  $attributeSetFactory
+     * @param EntityFactory               $entityFactory
+     * @param ProductFactory              $productFactory
+     * @param StockConfigurationInterface $stockConfiguration
+     * @param Processor                   $mediaGalleryProcessor
+     * @param ChildrenWeightResolver      $childrenWeightResolver
      */
     public function __construct(
         Type\Configurable $configurableProduct,
-        \Magento\Eav\Model\Entity\Attribute\SetFactory $attributeSetFactory,
-        \Magento\Eav\Model\EntityFactory $entityFactory,
-        \Magento\Catalog\Model\ProductFactory $productFactory,
-        \Magento\CatalogInventory\Api\StockConfigurationInterface $stockConfiguration,
-        \Magento\Catalog\Model\Product\Gallery\Processor $mediaGalleryProcessor
+        SetFactory $attributeSetFactory,
+        EntityFactory $entityFactory,
+        ProductFactory $productFactory,
```

Comment:
No need to revert but generally please avoid such changes mixed with change of logic. We will get rid of FQCN at some point but it should be a separate effort.